dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsQuery.java

```diff
@@ -42,44 +42,49 @@ public abstract class OracleStatsQuery {
   @Nonnull
   abstract String name();
 
-  @Nonnull
-  abstract String queryText();
-
   @Nonnull
   abstract StatsSource statsSource();
 
   @Nonnull
-  static OracleStatsQuery createAwr(String name, Duration queriedDuration) throws IOException {
+  static OracleStatsQuery createAwr(String name, Duration queriedDuration) {
     return create(false, queriedDuration, name, AWR);
   }
 
   @Nonnull
-  static OracleStatsQuery createNative(String name, boolean isRequired, Duration queriedDuration)
-      throws IOException {
+  static OracleStatsQuery createNative(String name, boolean isRequired, Duration queriedDuration) {
     return create(isRequired, queriedDuration, name, NATIVE);
   }
 
   @Nonnull
-  static OracleStatsQuery createStatspack(String name, Duration queriedDuration)
-      throws IOException {
+  static OracleStatsQuery createStatspack(String name, Duration queriedDuration) {
     return create(false, queriedDuration, name, STATSPACK);
   }
 
   private static OracleStatsQuery create(
-      boolean isRequired, Duration queriedDuration, String name, StatsSource statsSource)
-      throws IOException {
-    String path = String.format("oracle-stats/cdb/%s/%s.sql", statsSource.value, name);
-    return new AutoValue_OracleStatsQuery(
-        isRequired, queriedDuration, name, loadFile(path), statsSource);
+      boolean isRequired, Duration queriedDuration, String name, StatsSource statsSource) {
+    return new AutoValue_OracleStatsQuery(isRequired, queriedDuration, name, statsSource);
+  }
+
+  @Nonnull
+  String queryText() {
+    String source = statsSource().value;
+    String tenantSetup = "cdb";
+    String path = String.format("oracle-stats/%s/%s/%s.sql", tenantSetup, source, name());
+    return loadFile(path);
   }
 
   @Nonnull
   String description() {
     return String.format("Query{name=%s, statsSource=%s}", name(), statsSource());
   }
 
-  private static String loadFile(String path) throws IOException {
-    URL queryUrl = Resources.getResource(path);
-    return Resources.toString(queryUrl, UTF_8);
+  private static String loadFile(String path) {
+    try {
+      URL queryUrl = Resources.getResource(path);
+      return Resources.toString(queryUrl, UTF_8);
+    } catch (IOException e) {
+      String message = String.format("IOException caused by invalid file: %s.", path);
```

Comment:
Optional: I would suggest against using `IOException` in the error message, because it is duplicated information. It is already stated in the cause of IAE. The error message could simply be: "Error reading resource: '%s'.".

Reply:
Done, the message now says `"An invalid file was provided: [providedFileName]."`